File: app/src/main/java/adapter/CrewsAdapter.java
package adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

import activity.CrewsActivity;
import activity.PersonActivity;
import br.com.icaro.filme.R;
import info.movito.themoviedbapi.model.people.PersonCrew;
import utils.Constantes;
import utils.UtilsFilme;

/**
 * Created by icaro on 24/07/16.
 */
public class CrewsAdapter extends RecyclerView.Adapter<CrewsAdapter.CrewsViewHolder> {
    Context context;
    List<PersonCrew> crews;

    public CrewsAdapter(CrewsActivity crewsActivity, List<PersonCrew> crew) {

        this.context = crewsActivity;
        this.crews = crew;
        Log.d("CrewsAdapter", "Tamanho " + crews.size());
    }

    @Override
    public CrewsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.crews_list_adapter, parent, false);
        CrewsViewHolder viewHolder = new CrewsViewHolder(view);
        return viewHolder;
    }


    @Override
    public void onBindViewHolder(CrewsViewHolder holder, int position) {
        final PersonCrew personCrew = crews.get(position);
        holder.crew_character.setText(personCrew.getDepartment() + " " + personCrew.getJob());

        holder.crew_nome.setText(personCrew.getName());
        Picasso.with(context).load(UtilsFilme.getBaseUrlImagem(2) + personCrew.getProfilePath())
                .placeholder(R.drawable.person)
                .into(holder.img_crew);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(context, PersonActivity.class);
                intent.putExtra(Constantes.PERSON_ID, personCrew.getId());
                intent.putExtra(Constantes.NOME_PERSON, personCrew.getName());
                context.startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        Log.d("CrewsAdapter", "Tamanho " + crews.size());
        return crews.size() > 0 ? crews.size() : 0;
    }

    public class CrewsViewHolder extends RecyclerView.ViewHolder {

        TextView crew_nome, crew_character;
        ImageView img_crew;

        public CrewsViewHolder(View itemView) {
            super(itemView);
            crew_nome = (TextView) itemView.findViewById(R.id.crew_nome);
            crew_character = (TextView) itemView.findViewById(R.id.crew_character);
            img_crew = (ImageView) itemView.findViewById(R.id.img_crew);

        }
    }
}

File: app/src/main/java/adapter/CrewsAdapter.kt
package adapter

import activity.CrewsActivity
import activity.PersonActivity
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import br.com.icaro.filme.R
import com.squareup.picasso.MemoryPolicy
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import domain.CrewItem
import utils.Constantes
import utils.UtilsApp

/**
 * Created by icaro on 24/07/16.
 */
class CrewsAdapter(private val context: CrewsActivity, private val crews: List<CrewItem?>?) : RecyclerView.Adapter<CrewsAdapter.CrewsViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CrewsViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.crews_list_adapter, parent, false)
        return CrewsViewHolder(view)
    }


    override fun onBindViewHolder(holder: CrewsViewHolder, position: Int) {
        val crew = crews?.get(position)
        holder?.crew_character.text = crew?.department + " " + crew?.job

        holder.crew_nome.text = crew?.name
        Picasso.with(context)
                .load(UtilsApp.getBaseUrlImagem(UtilsApp.getTamanhoDaImagem(context, 2)) + crew?.profilePath)
                .error(R.drawable.person)
                .memoryPolicy(MemoryPolicy.NO_STORE, MemoryPolicy.NO_CACHE)
                .networkPolicy(NetworkPolicy.NO_CACHE, NetworkPolicy.NO_STORE)
                .into(holder.img_crew)

        holder.itemView.setOnClickListener {
            val intent = Intent(context, PersonActivity::class.java)
            intent.putExtra(Constantes.PERSON_ID, crew?.id)
            intent.putExtra(Constantes.NOME_PERSON, crew?.name)
            context.startActivity(intent)

        }

    }

    override fun getItemCount(): Int {
        return crews?.size ?: 0
    }

    inner class CrewsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

         val crew_nome: TextView
         val crew_character: TextView
         val img_crew: ImageView

        init {
            crew_nome = itemView.findViewById<View>(R.id.crew_nome) as TextView
            crew_character = itemView.findViewById<View>(R.id.crew_character) as TextView
            img_crew = itemView.findViewById<View>(R.id.img_crew) as ImageView

        }
    }
}

